clang/lib/Sema/JumpDiagnostics.cpp

```diff
@@ -658,7 +649,13 @@ void JumpScopeChecker::BuildScopeInformation(Stmt *S,
         Next = SC->getSubStmt();
       else if (LabelStmt *LS = dyn_cast<LabelStmt>(SubStmt))
         Next = LS->getSubStmt();
-      else
+      else if (AttributedStmt *AS = dyn_cast<AttributedStmt>(SubStmt)) {
+        if (GetMustTailAttr(AS)) {
+          LabelAndGotoScopes[AS] = ParentScope;
+          MustTailStmts.push_back(AS);
+        }
+        Next = AS->getSubStmt();
+      } else
```

Comment:
```suggestion
      else if (AttributedStmt *AS = dyn_cast<AttributedStmt>(SubStmt))
        Next = AS->getSubStmt();
      else
```
I think that the correct way to fix this issue is to leave the `case Stmt::AttributedStmtClass` above and to instead change this code to just get the substatement. Also update the comment above to mention that attributed statements are not "scope parents".

Reply:
Please correct me if my understanding is incorrect. 
This function is basically doing a DFS on the AST and skipping the nodes that is not considered to be scope parents(also could you confirm attributes can not be scope parents?). If my understanding is correct, the node for attributes will be skipped and ``case Stmt::AttributedStmtClass`` never gets executed.

Reply:
> also could you confirm attributes can never be scope parents?

I cannot think of an attribute which could be a scope parent. They're extra information associated with a statement (in this case), so the substatement needs to be walked.

> If my understanding is correct, the node for attributes will be skipped and case Stmt::AttributedStmtClass never gets executed.

Ah, I think you are correct! Sorry for the noise